Explain new Symfony kernel split (#1827)
PrestaShop/docs: modules/core-updates/9.0.md

- [Services translator, security.csrf.token_manager, session, session.flash_bag no longer public](https://github.com/PrestaShop/PrestaShop/pull/35069)
- [Replace AdminSecurity annotation with attributes](https://github.com/PrestaShop/PrestaShop/pull/35212)

### Kernel modifications

> Until PrestaShop 8.0 we relied on one Kernel that was used for the back office, in Symfony 9.0 we expanded the usage of the Symfony framework especially for two new features:

Reviewer: ```suggestion
Until PrestaShop 8.0, we relied on one Kernel used for the back office. In Symfony 9.0, we expanded the usage of the Symfony framework, especially for two new features:
```

> - the new Admin API based on ApiPlatform and OAuth authentication

Reviewer: ```suggestion
- the new Admin API based on API Platform and OAuth authentication
```

API Platform is an official name :D

Author: Yep you're right!

> - an experimental feature that gives access to a Symfony container on the frontend (in opposition to the custom built built container available today), this one is not very much used but it's a first stone for the future front migration

Reviewer: ```suggestion
- an experimental feature that gives access to a Symfony container in the front office (in opposition to the custom-built container available today). This one is not very much used, but it's a first stone for the future front office migration
```

> Those two new environments have different mechanisms and configuration from the back office, to separate these configurations cleanly they each have their dedicated kernel class and their own configuration. This allows to cleanly define different routing, security configurations, dedicated services, listeners, ...

Reviewer: ```suggestion
Those two new environments have mechanisms and configurations different from those in the back office. To separate these configurations cleanly, they each have their dedicated kernel class and their own configuration. This allows to cleanly define different routing, security configurations, dedicated services, listeners, ...
```

> Each kernel has a unique `applicationId` that allows to dynamize its configuration, its cache folder (since the services are not the same each kernel needs its own cache folder). The `appId` parameter has also been added globally to the `bin/console` tool. They all share the common `app/config/config.yml` configuration file but each one extends it in its own config folder.

Reviewer: ```suggestion
Each kernel has a unique `applicationId` that allows it to dynamize its configuration and cache folder (since the services are not the same, each kernel needs its own cache folder). The `appId` parameter has also been added globally to the `bin/console` tool. They all share the common `app/config/config.yml` configuration file, but each one extends it in its own config folder.
```

| Environment                 | Kernel class     | Config folder          | Cache folder                          | Endpoint               | Application ID |
|-----------------------------|------------------|------------------------|---------------------------------------|------------------------|----------------|
| Back office                 | `AdminKernel`    | `app/config/admin`     | `var/cache/{dev,prod,test}/admin`     | `/admin-dev/index.php` | `admin`        |
| Admin API                   | `AdminAPIKernel` | `app/config/admin-api` | `var/cache/{dev,prod,test}/admin-api` | `/admin-api/index.php` | `admin-api`    |
| Front office (experimental) | `FrontKernel`    | `app/config/front`     | `var/cache/{dev,prod,test}/front`     | `/index.php`           | `front`        |

{{% notice note %}}
**Console usage**

Clear the cache of the Admin API for its `prod` environment

`php bin/console cache:clear --env=prod --app-id=admin-api`

> Display event listeners for the Admin in `dev` environment (default value of `app-id` is `admin` for retro compatibility)

Reviewer: ```suggestion
Display event listeners for the back office `dev` environment (default value of `app-id` is `admin` for retro compatibility)
```
